bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -0,0 +1,219 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.websocket.client.ClientUpgradeRequest;
+import org.eclipse.jetty.websocket.client.WebSocketClient;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerService;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.yioremote.internal.YIOremoteBindingConstants.YIOREMOTEDOCKHANDLESTATUS;
+import org.openhab.binding.yioremote.internal.YIOremoteBindingConstants.YIOREMOTEMESSAGETYPE;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link YIOremoteDockHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Michael Loercher - Initial contribution
+ */
+@NonNullByDefault
+public class YIOremoteDockHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(YIOremoteDockHandler.class);
+
+    YIOremoteConfiguration localConfig = getConfigAs(YIOremoteConfiguration.class);
+    private WebSocketClient yioremoteDockHandlerwebSocketClient = new WebSocketClient();
+    private YIOremoteDockWebsocket yioremoteDockwebSocketClient = new YIOremoteDockWebsocket();
+    private ClientUpgradeRequest yioremoteDockwebSocketClientrequest = new ClientUpgradeRequest();
+    private @Nullable URI uriyiodockwebsocketaddress;
+    private YIOREMOTEDOCKHANDLESTATUS yioremotedockactualstatus = YIOREMOTEDOCKHANDLESTATUS.UNINITIALIZED;
+    private @Nullable Future<?> pollingJob;
+    private String sendircode = "";
+
+    public YIOremoteDockHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        scheduler.execute(() -> {
+            try {
+                uriyiodockwebsocketaddress = new URI("ws://" + localConfig.host + ":946");
+                yioremotedockactualstatus = YIOREMOTEDOCKHANDLESTATUS.AUTHENTICATIONPROCESS;
+                try {
+                    yioremoteDockHandlerwebSocketClient.start();
+                } catch (Exception e) {
+                    logger.warn("Web socket start failed {}", e.getMessage());
+                    // throw new IOException("Web socket start failed");
+                }
+                try {
+                    yioremoteDockHandlerwebSocketClient.connect(yioremoteDockwebSocketClient,
+                            uriyiodockwebsocketaddress, yioremoteDockwebSocketClientrequest);
+                    yioremoteDockwebSocketClient.getLatch().await();
```

Comment:
What happens if the connection never gets established? As this is a thread of a shared thread pool, it shouldn't block for a long time. A better approach would be to execute the code below from within `onConnect()`.

Reply:
solved